### PR TITLE
Opensearch Client (PR 3 of 3)

### PR DESCRIPTION
**Pull request by JenniWhitman**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
updates https://github.com/mitodl/open-discussions/issues/3494

#### What's this PR do?
This PR swaps the Elasticsearch client out for OpenSearch. This will run the OpenSearch client against the Elasticsearch service.  The next and final PR in the series will update the service.

#### How should this be manually tested?
Pull the branch locally and run docker build and docker up to change your client to OpenSearch. The Docker container (service) will not change.

I would then:

run tests
check that you can use the search API manually using postman or fetch or some tool of your choice - specifically as OCW might make requests
make sure that pages within Open that are using the search API function (new open demo, infinite, learn, open discussions)

#### Where should the reviewer start?
Pull the branch and play with it, honestly. I can run all the tests in the world and I can try to break it, but ultimately it's best if we have many people poke it from all sides.

#### Any background context you want to provide?
Testing in a similar manner to the ES7 PR would probably be a good start!

### CODE REVIEW

**Comment by JenniWhitman** [comment]
Anastasia surfaced two issues while testing:
1. The index was removed on build. This appears to be because I added `container name` to the docker-compose (which we will need for OpenSearch container in next PR) which made it build a new container. This won't be a problem in prod because Mike will be doing a migration and not a new install, but on our docker containers, we see this behavior. Testing, but should be resolved.
2. `recreate_index` is failing after I merged `master` in - this goes back to the mapping issues previous which I've set to be the same as our 7.10 branch, which should behave precisely as OS 1.0.0 does, and still things are not working so I am investigating

**Review by abeglova — APPROVED** [review]
lgtm

**Comment by JenniWhitman** [comment]
Putting this as do not merge until salt is in

File: docker-compose.yml
Comment (JenniWhitman) [reply docker-compose.yml:16]: yes until the service changes

**Pull request merged**
